### PR TITLE
SCUMM: Reuse bomp decoder for SMUSH RLE codec

### PR DESCRIPTION
**Pull request by BLooperZ**
This refactors SMUSH RLE codec (previously codec1) to use BOMP decoder (which is the same algorithm)
Added a flag to BOMP decoder to support transparency (another option was to zero-out the buffer before decoding).

There was a slight difference in the old implementation, of using the source length instead of the given width,
I haven't seen any difference when batch decoding all graphics from FT/DIG/COMI/Mortimer back when I implemented it in python.
anyway, judging by the last line in old implementation* (`dst += pitch - left - width;`), it could potentially cause graphic glitches if for some reason the source line stopped prematurely. (might be a good idea to verify this on disasm, and if needed add both conditions)

I have verified FT's mineroad looks ok.

* the `- width` is apparently needed to cancel the increments to `dst` along the way, which potentially might not match.

### CODE REVIEW

File: engines/scumm/bomp.cpp [hunk 1]
```diff
@@ -64,12 +64,17 @@ void bompDecodeLine(byte *dst, const byte *src, int len) {
 		len -= num;
 		if (code & 1) {
 			color = *src++;
-			memset(dst, color, num);
+			if (setZero || color)
+				memset(dst, color, num);
+			dst += num;
 		} else {
-			memcpy(dst, src, num);
```
Comment (ccawley2011) [inline engines/scumm/bomp.cpp:69]: It might be better to continue using `memcpy` when `setZero` is true.

Comment (BLooperZ) [reply engines/scumm/bomp.cpp:69]: thank you, done

File: engines/scumm/smush/codec1.cpp [hunk 2]
```diff
@@ -22,40 +22,18 @@
...
-				lineSize -= length;
-				while (length--) {
-					val = *src++;
-					if (val)
-						*dst = val;
-					dst++;
-				}
-			}
-		}
-		dst += pitch - left - width;
```
Comment (AndywinXp) [inline engines/scumm/smush/codec1.cpp:57]: Well, I haven't seen references of `- width` in the disasm... but then again, my knowledge of that part is very modest. The whole line (with `pitch` and `left`) was added by sev about 17 years ago. I guess it had a purpose back then. But now I really can't tell, and I couldn't see any differences playing FT for a while on the current master build and your build...

**Review by AndywinXp — APPROVED** [review]
Thanks for these changes! I have added a minor comment which requires no changes, but would maybe require asking sev about something he wrote 17 years ago... but given I don't even remember code I wrote 1 year ago, that's a really long shot 😂

**Pull request merged**